low_level_platform/impl/src/lf_patmos_support.c

```diff
@@ -0,0 +1,85 @@
+#include <time.h>
+#include <errno.h>
+#include <assert.h>
+#include <lf_patmos_support.h>
+#include "../platform.h"
+#include <machine/rtc.h>
+#include <machine/exceptions.h>
+#include <stdio.h>
+
+// Keep track of physical actions being entered into the system
+static volatile bool _lf_async_event = false;
+// Keep track of whether we are in a critical section or not
+static volatile int _lf_num_nested_critical_sections = 0;
+/**
+ * @brief Sleep until an absolute time.
+ * TODO: For improved power consumption this should be implemented with a HW timer and interrupts.
```

Comment:
This TODO is probably out-of-date. A relevant question is how this sleeping can be done better for Patmos. It is not ideal to busy wait. Is there any sleeping instructions or wait-fo-interrupt instructions?

Reply:
Since there is no sleep mode in Patmos, and energy saving is not important for real-time systems, we just used a busy sleep. So I'll remove this TODO and put this explanation instead.